apps/web/src/features/spaces/components/SpaceAddressBook/EditContactDialog.tsx

```diff
@@ -0,0 +1,189 @@
+import useChains from '@/hooks/useChains'
+import type { ContactField } from './AddContact'
+
+type EditContactDialogProps = {
+  entry: SpaceAddressBookEntry
+  onClose: () => void
+}
+
+// TODO: Replace with actual CGW API calls
+const createAddressBookEntry = async (address: string, name: string, chainId: string) => {
+  console.log('CREATE:', { address, name, chainId })
+  return
+}
+
+const updateAddressBookEntry = async (id: string, name: string) => {
+  console.log('UPDATE:', { id, name })
+  return
+}
+
+const deleteAddressBookEntry = async (id: string, chainId: string) => {
+  console.log('DELETE:', { id, chainId })
+  return
+}
+
+const EditContactDialog = ({ entry, onClose }: EditContactDialogProps) => {
+  const [error, setError] = useState('')
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const { configs } = useChains()
+
+  const defaultNetworks = entry.networks
+    .map((network) => {
+      return configs.find((chain) => chain.chainId === network.chainId) as ChainInfo
+    })
+    .filter(Boolean)
+
+  const defaultValues = {
+    name: entry.name,
+    address: entry.address,
+    networks: defaultNetworks,
+  }
+
+  const methods = useForm<ContactField>({
+    mode: 'onChange',
+    defaultValues,
+  })
+
+  const { handleSubmit, formState, control, reset, watch } = methods
+
+  const { errors } = formState
+
+  // Watch for changes in name and networks
+  const watchedName = watch('name')
+  const watchedNetworks = watch('networks')
+
+  // Check if any changes were made
+  const hasChanges = useMemo(() => {
+    const nameChanged = watchedName !== entry.name
+
+    const originalChainIds = entry.networks.map((network) => network.chainId).sort()
+    const currentChainIds = watchedNetworks.map((network) => network.chainId).sort()
+    const networksChanged =
+      originalChainIds.length !== currentChainIds.length ||
+      originalChainIds.some((id, index) => id !== currentChainIds[index])
+
+    return nameChanged || networksChanged
+  }, [watchedName, watchedNetworks, entry.name, entry.networks])
+
+  const handleClose = () => {
+    reset(defaultValues)
+    setError('')
+    onClose()
+  }
+
+  const onSubmit = handleSubmit(async (data) => {
+    try {
+      setIsSubmitting(true)
+      trackEvent({ ...SPACE_EVENTS.ADD_ADDRESS_SUBMIT })
+
+      const selectedChainIds = data.networks.map((network) => network.chainId)
+      const originalChainIds = entry.networks.map((network) => network.chainId)
+
+      // Determine which networks need to be created, updated, or deleted
+      const toCreate = selectedChainIds.filter((chainId) => !originalChainIds.includes(chainId))
+      const toDelete = originalChainIds.filter((chainId) => !selectedChainIds.includes(chainId))
+      // Only update if the name has changed
+      const nameHasChanged = data.name !== entry.name
+      const toUpdate = nameHasChanged ? selectedChainIds.filter((chainId) => originalChainIds.includes(chainId)) : []
```

Comment:
It feels a bit complex to combine these two actions into one form i.e. if a user deselects networks or adds more networks and also changes the name we would have to delete/add some entries while updating the name in others. IMO it would be easier to only allow changing the name. If a multi-network entry is edited we would update each entry with that new name and if its a contact on a single network we just update one name.

Reply:
After our discussion today we decided to keep it this way so it will be possible to add/remove networks when editing an address and there will be one endpoint where we just have to send all selected networks and it will do the upsertion so I think this good.